packages/caravan-clients/src/client.ts

```diff
@@ -637,6 +637,228 @@ export class BlockchainClient extends ClientBase {
     }
   }
 
+/**
+ * Retrieves transaction history for one or more addresses (public clients only)
+ * 
+ * This method is designed for public blockchain explorers (Mempool/Blockstream) that
+ * maintain address indexes, allowing efficient address-specific queries.
+ * 
+ * Why this method is PUBLIC CLIENT ONLY:
+ * - Public APIs maintain address indexes for efficient lookups
+ * - Bitcoin Core doesn't support address-specific queries (see getWalletTransactionHistory)
+ * - Allows querying multiple addresses with Promise.all for efficiency
+ * 
+ * For PRIVATE clients: Use getWalletTransactionHistory() which returns all wallet transactions
+ * 
+ * @param address - Single address or array of addresses to query
+ * @param count - Number of transactions to return per address (1-100)
+ * @param skip - Number of transactions to skip for pagination
+ * @returns Combined array of transactions sorted by time (newest first)
+ * @throws Error if called on private client
+ * 
+ * @example
+ * // Single address
+ * const txs = await client.getAddressTransactionHistory("bc1q...");
+ * 
+ * // Multiple addresses
+ * const txs = await client.getAddressTransactionHistory(["bc1q...", "bc1p..."]);
+ * 
+ * @see getWalletTransactionHistory - For private clients
+ */
+
+  public async getAddressTransactionHistory(
+  address: string | string[],
+  count: number = 10,
+  skip: number = 0
+): Promise<TransactionDetails[]> {
+  if (count < 1 || count > 100000) {
+    throw new Error("Count must be between 1 and 100000");
+  }
+  if (skip < 0) {
+    throw new Error("Skip must be non-negative");
+  }
+
+  try {
+    if (this.type === ClientType.PRIVATE) {
+      throw new Error("Use getWalletTransactionHistory for private clients");
+    }
+
+    if (!this.provider) {
+      throw new Error("Provider must be specified for public clients");
+    }
+
+    const addresses = Array.isArray(address) ? address : [address];
+    const allTransactions: any[] = [];
+    let anyAddressFailed = false; // Track if any request failed
+    let firstError: Error | null = null; // Track first error
+
+    // Use allSettled to handle partial failures
+    const results = await Promise.allSettled(
+      addresses.map(async (addr) => {
+        try {
+          const query = this.provider === PublicBitcoinProvider.BLOCKSTREAM
+            ? `limit=${count}&offset=${skip}`
+            : `count=${count}&skip=${skip}`;
+          const endpoint = `/address/${addr}/txs?${query}`;
+          const response = await this.Get(endpoint);
+          
+          // Robust response handling
+          let transactions: any[] = [];
+          
+          // Case 1: Response is already an array
+          if (Array.isArray(response)) {
```

Comment:
Are we testing the handling of different response shapes and the transformations that might be done in them?